runhouse/resources/hardware/cluster.py

```diff
@@ -1128,6 +1128,24 @@ def _start_or_restart_helper(
         # Update the cluster config on the cluster
         self.save_config_to_cluster()
 
+        # Save the local ~/.rh config to the cluster with the user's hashed token,
+        # if such does not exist on the cluster
+        if rns_client.token:
```

Comment:
why are we branching here, i dont think we had this previously? if the token deosn't exist we shouldn't need to sync anything

Reply:
actually we had it previously, we created the config.yaml only if there is a rh token. (same goes with the new implementation)